### PR TITLE
Define the CHIDs in a more generic way

### PR DESCRIPTION
**Pull request by hughsie**
Also, add a (nonstandard) CHID GUIDs to make `FWUPD_PLUGIN_FLAG_REQUIRE_HWID` more useful.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — COMMENTED** [review]
Yeah I like this approach better

**Review by superm1 — DISMISSED** [review]

File: libfwupdplugin/fu-hwids.c [hunk 1]
```diff
@@ -422,7 +426,51 @@ fu_hwids_init(FuHwids *self)
...
+	fu_hwids_add_chid(self, "HardwareID-11", "Manufacturer&Family");
+	fu_hwids_add_chid(self, "HardwareID-12", "Manufacturer&EnclosureKind");
+	fu_hwids_add_chid(self,
+			  "HardwareID-13",
+			  "Manufacturer&BaseboardManufacturer&BaseboardProduct");
+	fu_hwids_add_chid(self, "HardwareID-14", "Manufacturer");
+
+	/* used by the flashrom plugin */
+	fu_hwids_add_chid(self,
+			  "HardwareID-01-without-BiosVersion-nonstd",
```
[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/fu-hwids.c:471]: I went back and forwards on this; I wanted to tie it back to the `HardwareID` it was based on. Maybe something like `nonstdHardwareID-01` or `HardwareID-01.01`

[the remark this answers is not in the document]
Comment (superm1) [reply libfwupdplugin/fu-hwids.c:471]: Does it matter what it's based on? As long as you can easily decode what it means it is just a bullet in some docs

[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/fu-hwids.c]: Maybe we can do both, perhaps like below?

**Review by superm1 — APPROVED** [review]

**Pull request merged**